ddp-workspace/projects/ddp-dsm-ui/src/app/file-download/file-download.component.ts

```diff
@@ -21,47 +22,79 @@ export class FileDownloadComponent implements OnDestroy {
   downloading = false;
   CLEAN = 'CLEAN';
   subscription: Subscription;
+  subscription2: Subscription;
 
-  constructor(private dsmService: DSMService ) {
+  constructor( private dsmService: DSMService ) {
 
   }
 
   ngOnDestroy(): void {
     if (this.subscription != null) {
       this.subscription.unsubscribe();
     }
+    if (this.subscription2 != null) {
+      this.subscription2.unsubscribe();
+    }
   }
 
   downloadParticipantFile( file: File ): void {
     this.downloading = true;
-    if (file.scanResult !== this.CLEAN) {
+    if (!this.isFileClean( file )) {
       this.downloadMessage = 'Error - file has not passed scanning';
       this.downloading = false;
       return;
     }
     const realm = localStorage.getItem( ComponentService.MENU_SELECTED_REALM );
-    this.subscription = this.dsmService.downloadParticipantFile( file.fileName, file.bucket, file.blobName, realm, file.mimeType )
-      .subscribe( {
+    this.subscription = this.dsmService.getSignedUrl( this.participant.data.profile[ 'guid' ], file.fileName, file.bucket,
+      file.blobName, file.guid, realm ).subscribe( {
         next: data => {
-          this.saveParticipantFile( data, file.mimeType, file.fileName );
+          this.saveParticipantFile( data.url, file.mimeType, data.fileName );
+        },
+        error: err => {
+          this.downloadMessage = err;
+          this.downloading = false;
```

Comment:
The block 

```
this.downloadMessage = some message;
this.downloading = false;
```
is used 4 times in the file.

Perhaps, it is worth extracting it in as a separate method?